src/main/java/org/spongepowered/api/resource/Resource.java

```diff
@@ -0,0 +1,178 @@
+ *         logger.info(res.readString(DefaultCharsets.UTF_8));
+ *     } catch (IOException e) {
+ *         logger.warn("Failed to load resource: {}", path, e);
+ *     }
+ * </pre>
+ */
+public interface Resource extends Closeable {
+
+    /**
+     * Returns a new {@link InputStream} of this resource. A new input stream
+     * should be created each time this method is called.
+     *
+     * @return A new input stream
+     */
+    InputStream getInputStream();
+
+    /**
+     * Gets the metadata for this resource.
+     *
+     * <p>The metadata file has the same name as this resource, but has
+     * {@code .mcmeta} appended to the end.</p>
+     *
+     * <p>For example: the metadata for the resource
+     * {@code minecraft:textures/blocks/water_flow.png} would be located at
+     * {@code minecraft:textures/blocks/water_flow.png.mcmeta}</p>
+     *
+     * @return The metadata or {@link Optional#empty() empty} if it doesn't exist.
+     * @see <a href=http://minecraft.gamepedia.com/Resource_pack#Contents> Minecraft Wiki/Resource Packs
+     */
+    Optional<DataView> getMetadata();
+
+    /**
+     * Gets the path of this resource.
+     *
+     * @return The path
+     */
+    ResourcePath getPath();
+
+    /**
+     * Gets the pack name which owns this resource.
+     *
+     * @return The parent pack.
+     */
+    String getPackName();
+
+    /**
+     * Gets a reader for this resource using the given {@link Charset}.
+     *
+     * @param charset The charset
+     * @return A new reader
+     */
+    default BufferedReader getReader(Charset charset) {
+        return new BufferedReader(new InputStreamReader(getInputStream(), charset));
+    }
+
+    /**
+     * Reads this resource into a string using the given charset.
+     *
+     * @param charset The charset
+     * @return The string contents
+     * @throws IOException if an error occurs
+     */
+    default String readString(Charset charset) throws IOException {
+        return CharStreams.toString(getReader(charset));
+    }
+
+    /**
+     * Reads a list of string lines from this resource using the given charset.
+     *
+     * @param charset The charset
+     * @return The list of strings
+     */
+    default Stream<String> readLines(Charset charset) {
+        return getReader(charset).lines();
+    }
+
+    /**
+     * Reads the bytes from this resource and returns them in an array.
+     *
+     * @return The bytes
+     * @throws IOException if an error occurs
+     */
+    default byte[] readBytes() throws IOException {
+        return ByteStreams.toByteArray(getInputStream());
+    }
+
+    /**
+     * Reads this resource into a {@link DataView}.
+     *
+     * @param format The data format to use
+     * @return The dataview
+     * @throws IOException if an error occurs
+     */
+    default DataView readDataView(DataFormat format) throws IOException {
+        try {
+            return format.readFrom(getInputStream());
```

Comment:
Possible resource leak; unfortunately sponge api did not document if the stream will be closed after usage.